### ISSUE TITLE
DisplayServer: Instant disappearance of popup submenu if using keyboard while pointer is on the parent

### ISSUE THREAD

**Issue by Houkime** [godotengine/godot#50529]
### Godot version

v4.0.dev.custom_build [a0d800e96]
### System information

Arch Linux (kernel 5.12.15-arch1-1, X11), WM: at least on dwm and LXQt

### Issue description

While investigating https://github.com/godotengine/godot/issues/50367, i found another bug.

When the mouse pointer is on a popup menu, if one uses keyboard to navigate to a submenu (arrow keys), it opens but closes quickly.
This does not happen when the pointer is not on the parent body.
This happens at the very least on LXQt and dwm.
This DOES happen in single window mode. Even worse, in a single window mode, it happens no matter where your pointer is. Also, in the single window mode this happens with mouse clicks to open subwindows too.

On gif, i first try navigating with the pointer not on the parent body (it sort of works, though quirky), and then with the pointer on it.
![submenu_keyboard_bug](https://user-images.githubusercontent.com/32805122/126003732-62594374-6ded-496d-a2d0-aa08d4875e7f.gif)

debug output (lxqt, right after the right arrow is pushed to open a submenu, multiple window mode)

```
about_to_popup
 visibility set to true
[6121] MapNotify window=41943265 (10)
requesting focus because menu window is started(MapNotify)
[6121] VisibilityNotify window=41943265 (10), state=0
[6121] Expose window=41943265 (10), count='0'
[6122] FocusOut window=41943185 (2), mode='0'
[6122] FocusIn window=41943265 (10), mode='0'
hide was pushed
 visibility set to false
 from window_set_transient calling focus on (2) as a prev_parent of (10) (This is probably unrelated, since the hide was issued before that)
delete_sub_window: 41943265 (10)
```
And the backtrace of the hide push. 
The signal at stack frame 12 is a timeout signal, and it is issued by a minimum lifetime timer set on 0.3 seconds.

```gdb
Thread 1 "godot.linuxbsd." hit Breakpoint 1, MessageQueue::push_call (this=0x98c1a30,
    p_id=..., p_method=..., p_args=0x7fffffffd5e0, p_argcount=0, p_show_error=false)
    at core/object/message_queue.cpp:48
48	        print_line("hide was pushed");
(gdb) backtrace
#0  MessageQueue::push_call (this=0x98c1a30, p_id=..., p_method=...,
    p_args=0x7fffffffd5e0, p_argcount=0, p_show_error=false)
    at core/object/message_queue.cpp:48
#1  0x0000000005d29bf3 in MessageQueue::push_call (this=0x98c1a30, p_id=...,
    p_method=..., p_arg1=..., p_arg2=..., p_arg3=..., p_arg4=..., p_arg5=...)
    at core/object/message_queue.cpp:66
#2  0x0000000005d2a95b in MessageQueue::push_call (this=0x98c1a30,
    p_object=0x10ebb2e0, p_method=..., p_arg1=..., p_arg2=..., p_arg3=..., p_arg4=...,
    p_arg5=...) at core/object/message_queue.cpp:124
#3  0x0000000005d3c0ea in Object::call_deferred (this=0x10ebb2e0, p_method=...,
    p_arg1=..., p_arg2=..., p_arg3=..., p_arg4=..., p_arg5=...)
    at core/object/object.cpp:1650
#4  0x0000000004623210 in Popup::_close_pressed (this=0x10ebb2e0)
    at scene/gui/popup.cpp:106
#5  0x000000000462eb67 in PopupMenu::_close_pressed (this=0x10ebb2e0)
    at scene/gui/popup_menu.cpp:684
#6  0x000000000462eaf1 in PopupMenu::_minimum_lifetime_timeout (this=0x10ebb2e0)
    at scene/gui/popup_menu.cpp:669
#7  0x00000000046481fb in call_with_variant_args_helper<PopupMenu>(PopupMenu*, void (PopupMenu::*)(), Variant const**, Callable::CallError&, IndexSequence<>) (
    p_instance=0x10ebb2e0,
    p_method=(void (PopupMenu::*)(PopupMenu * const)) 0x462ea72 <PopupMenu::_minimum_lifetime_timeout()>, p_args=0x7fffffffdbf0, r_error=...)
    at ./core/variant/binder_common.h:218
#8  0x0000000004646c87 in call_with_variant_args<PopupMenu> (p_instance=0x10ebb2e0,
    p_method=(void (PopupMenu::*)(PopupMenu * const)) 0x462ea72 <PopupMenu::_minimum_lifetime_timeout()>, p_args=0x7fffffffdbf0, p_argcount=0, r_error=...)
    at ./core/variant/binder_common.h:332
#9  0x0000000004644dad in CallableCustomMethodPointer<PopupMenu>::call (
    this=0x10ec7fc0, p_arguments=0x7fffffffdbf0, p_argcount=0, r_return_value=...,
    r_call_error=...) at ./core/object/callable_method_pointer.h:96
#10 0x0000000005a99eaf in Callable::call (this=0xb8d66e8, p_arguments=0x7fffffffdbf0,
    argcount=0, r_return_value=..., r_call_error=...) at core/variant/callable.cpp:50
#11 0x0000000005d34c06 in Object::emit_signal (this=0x10ec7db0, p_name=..., p_args=0x7fffffffdbf0, p_argcount=0) at core/object/object.cpp:1099
#12 0x0000000005d35318 in Object::emit_signal (this=0x10ec7db0, p_name=..., p_arg1=..., p_arg2=..., p_arg3=..., p_arg4=..., p_arg5=...) at core/object/object.cpp:1154
#13 0x0000000004450165 in Timer::_notification (this=0x10ec7db0, p_what=25) at scene/main/timer.cpp:61
#14 0x0000000004452980 in Timer::_notificationv (this=0x10ec7db0, p_notification=25, p_reversed=false) at scene/main/timer.h:37
#15 0x0000000005d32dd3 in Object::notification (this=0x10ec7db0, p_notification=25, p_reversed=false) at core/object/object.cpp:841
#16 0x000000000443071e in SceneTree::_notify_group_pause (this=0xb513410, p_group=..., p_notification=25) at scene/main/scene_tree.cpp:842
#17 0x000000000442e87f in SceneTree::process (this=0xb513410, p_time=0.00666666729) at scene/main/scene_tree.cpp:448
#18 0x00000000020ba665 in Main::iteration () at main/main.cpp:2533
#19 0x0000000002078888 in OS_LinuxBSD::run (this=0x7fffffffe0f0) at platform/linuxbsd/os_linuxbsd.cpp:278
#20 0x00000000020757a4 in main (argc=3, argv=0x7fffffffe5e8) at platform/linuxbsd/godot_linuxbsd.cpp:58
```
The lifetime timer creation and connection code is around popup_menu.cpp:1722
https://github.com/godotengine/godot/blob/master/scene/gui/popup_menu.cpp#L1718

```
minimum_lifetime_timer = memnew(Timer);
minimum_lifetime_timer->set_wait_time(0.3);
minimum_lifetime_timer->set_one_shot(true);
minimum_lifetime_timer->connect("timeout", callable_mp(this, &PopupMenu::_minimum_lifetime_timeout));
add_child(minimum_lifetime_timer);
```

### Steps to reproduce

* Create a new project
* Click on one of the topbar items (it opens a menu)
* Move pointer onto the popup body 
* Using arrow keys, open any submenu

With single window mode, it does not matter whether your pointer is at the popup or not. Also, in single window mode this happens with mouse clicks to open subwindows too.

### Minimal reproduction project

New empty project is good enough.

**Comment by pouleyKetchoupp** [comment]
I can confirm the bug, but it's not specific to X11. I've reproduced it on both ArchLinux/Gnome and Windows 10.

**Comment by KoBeWi** [comment]
Partially fixed in 83d2673
The menu doesn't disappear now, but you can't open sub-menus using keyboard.
